crates/autopilot/src/run_loop.rs

```diff
@@ -74,14 +79,81 @@ impl RunLoop {
 
         // Shuffle so that sorting randomly splits ties.
         solutions.shuffle(&mut rand::thread_rng());
-        solutions.sort_unstable_by(|left, right| left.1.score.total_cmp(&right.1.score));
+        solutions.sort_unstable_by_key(|solution| solution.1.score);
 
         // TODO: Keep going with other solutions until some deadline.
         if let Some((index, solution)) = solutions.pop() {
             // The winner has score 0 so all solutions are empty.
-            if solution.score == 0. {
+            if solution.score == 0.into() {
                 return;
             }
+
+            let auction_id = id;
+            let winner = solution.submission_address;
+            let winning_score = solution.score;
+            let reference_score = solutions
+                .last()
+                .map(|(_, response)| response.score)
+                .unwrap_or_default();
+            let mut participants = solutions
+                .iter()
+                .map(|(_, response)| response.submission_address)
+                .collect::<HashSet<_>>();
+            participants.insert(solution.submission_address); // add winner as participant
+            let prices = auction.prices.clone();
```

Comment:
Here specifically, I would:

```rust
let mut prices = BTreeMap::new();
```

And then, in the orders loop (in pseudocode):
```rust
prices.insert(auction_order.sell_token, auction.prices[auction_order.sell_token]);
prices.insert(auction_order.buy_token, auction.prices[auction_order.buy_token]);
```